stdlib/public/core/IntegerParsing.swift

```diff
@@ -192,9 +95,362 @@ extension FixedWidthInteger {
   ///
   /// - Parameter description: The ASCII representation of a number.
   @inlinable
-  @_semantics("optimize.sil.specialize.generic.partial.never")
-  @inline(__always)
   public init?(_ description: String) {
     self.init(description, radix: 10)
   }
 }
+
+@usableFromInline
+internal func _parseIntegerBase10(
+  from utf8: UnsafeBufferPointer<UInt8>, allowNegative: Bool, max: UInt64
+) -> UInt64? {
+  return _parseIntegerSpecialized(from: utf8, allowNegative: allowNegative,
+                                  max: max, using: _parseUnsignedBase10(from:))
+}
+
+@usableFromInline
+internal func _parseIntegerBase16(
+  from utf8: UnsafeBufferPointer<UInt8>, allowNegative: Bool, max: UInt64
+) -> UInt64? {
+  return _parseIntegerSpecialized(from: utf8, allowNegative: allowNegative,
+                                  max: max, using: _parseUnsignedBase16(from:))
+}
+
+@usableFromInline
+internal func _parseIntegerBase2(
+  from utf8: UnsafeBufferPointer<UInt8>, allowNegative: Bool, max: UInt64
+) -> UInt64? {
+  return _parseIntegerSpecialized(from: utf8, allowNegative: allowNegative,
+                                  max: max, using: _parseUnsignedBase2(from:))
+}
+
+@usableFromInline
+internal func _parseInteger(
+  from utf8: UnsafeBufferPointer<UInt8>,
+  radix: Int,
+  allowNegative: Bool,
+  max: UInt64
+) -> UInt64? {
+  return _parseIntegerSpecialized(from: utf8, allowNegative: allowNegative,
+                                  max: max, using: { utf8 in
+    var iter = utf8.makeIterator()
+    let first = iter.next()._unsafelyUnwrappedUnchecked
+    return _parseFromUTF8Unsigned(first: first, rest: &iter, radix: radix)
+  })
+}
+
+@inline(__always)
```

Comment:
I'd like to see the benchmark results if this function and the following `_parseUnsignedBaseXX` functions were all `@usableFromInline` instead of `@inline(__always)`.  Could you try that experiment?  I suspect that you would still see a healthy speedup over the old version with dramatically smaller code size in the clients.

Reply:
Note that these functions are not ABI-public, they are never inlined in the client only in the standard library itself. They are just a convenience to generate the specialized @usableFromInline methods above.

Reply:
Ah.  Of course, that makes perfect sense.  Might be worth checking whether these annotations are really needed; I'm not an expert in compiler optimizations, but I have the impression that it does a pretty good job of inlining within a module.